development/workflow.rst

```diff
@@ -0,0 +1,305 @@
+Following :ref:`our best practices <git-commit-organization-best-practices>`, any fix-up commits should be squashed into the appropriate original commit to present a clean development story.
+
+Once the iterative review process is complete, the reviewer should switch the JIRA ticket's state to **Reviewed**.
+
+Merging
+-------
+
+Putting a ticket in a **Reviewed** state gives the developer the go-ahead to merge the ticket branch.
+If it has not been done already, the develop should rebase the ticket branch against the latest master.
+If a rebase was required, a final ticket branch check with Jenkins should be done.
+
+To merge a ticket branch, we always use non-fast forward merges so that the merge point is marked in the commit history.
+
+.. code-block:: bash
+
+   git checkout master
+   git pull  # sanity check
+   git merge --no-ff tickets/DM-NNNN
+   git push
+
+Once the merge has been completed, the developer should mark the JIRA ticket as **Done**.
+
+.. _git-setup-best-practices:
+
+Appendix: Git and Github Setup Suggestions
+==========================================
+
+.. _git-github-2fa:
+
+Setup Two-Factor Authentication (2FA) for GitHub
+------------------------------------------------
+
+We encourage you to enable `Two-Factor Authentication (2FA) for GitHub <https://help.github.com/articles/about-two-factor-authentication/>`_ through your `account security settings <https://github.com/settings/security>`_.
+2FA means that you'll have to enter an authentication code when logging into GitHub.com from a new computer.
+Apps like `1Password <https://agilebits.com/onepassword>`_ (see their `guide <https://guides.agilebits.com/1password-ios/5/en/topic/setting-up-one-time-passwords>`_), `Authy <https://www.authy.com>`_, and the Google Authenticator App can help you generate these authentication codes.
+When pushing code with a 2FA-enabled account, you'll use a personal access token instead of your password.
+You can `create and revoke access easily from your GitHub settings page <https://github.com/settings/tokens>`_.
+To help you automatically authenticate when pushing to GitHub, we encourage you to follow the next step and enable a credential helper.
+
+.. _git-credential-helper:
+
+Setup a Git Credential Helper
+-----------------------------
+
+Rather than entering your GitHub username and password (or 2FA access token) every time you push, you can setup a Git credential helper to manage this for you.
+A credential helper is especially important for working with our :doc:`Git LFS-backed repositories </development/git_lfs>`.
+
+**Mac users** can use the secure OS X keychain:
+
+.. code-block:: bash
+
+   git config --global credential.helper osxkeychain  
+
+**Linux users** can use a credential *cache* to temporarily keep your credentials in memory.
+To have your credentials cached for 1 hour (3600 seconds):
+
+.. code-block:: bash
+
+   git config --global credential.helper 'cache --timeout=3600'
```

Comment:
If your home directory is on a network disk (NFS, AFS), you need to do a bit more.  See Git LFS docs.